plugins/woocommerce/tests/php/src/Internal/DataStores/Orders/OrdersTableRefundDataStoreTests.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * File containing the class WP_Test_WC_Order_Refund.
+ */
+
+use Automattic\WooCommerce\Database\Migrations\CustomOrderTable\PostsToOrdersMigrationController;
+use Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore;
+use Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableRefundDataStore;
+use Automattic\WooCommerce\RestApi\UnitTests\Helpers\OrderHelper;
+
+/**
+ * Class OrdersTableRefundDataStoreTests.
+ */
+class OrdersTableRefundDataStoreTests extends WC_Unit_Test_Case {
+
+	/**
+	 * @var PostsToOrdersMigrationController
+	 */
+	private $migrator;
+
+	/**
+	 * @var OrdersTableRefundDataStore
+	 */
+	private $sut;
+
+	/**
+	 * @var OrdersTableDataStore;
+	 */
+	private $order_data_store;
+
+	/**
+	 * @var WC_Order_Refund_Data_Store_CPT
+	 */
+	private $cpt_data_store;
+
+	/**
+	 * Initializes system under test.
+	 */
+	public function setUp(): void {
+		parent::setUp();
+		// Remove the Test Suite’s use of temporary tables https://wordpress.stackexchange.com/a/220308.
+		remove_filter( 'query', array( $this, '_create_temporary_tables' ) );
+		remove_filter( 'query', array( $this, '_drop_temporary_tables' ) );
+		OrderHelper::delete_order_custom_tables(); // We need this since non-temporary tables won't drop automatically.
+		OrderHelper::create_order_custom_table_if_not_exist();
+		$this->sut              = wc_get_container()->get( OrdersTableRefundDataStore::class );
+		$this->order_data_store = wc_get_container()->get( OrdersTableDataStore::class );
+		$this->migrator         = wc_get_container()->get( PostsToOrdersMigrationController::class );
+		$this->cpt_data_store   = new WC_Order_Refund_Data_Store_CPT();
+	}
+
+	/**
+	 * Test that we are able to create refund.
+	 */
+	public function test_create_refund() {
```

Comment:
This test seems A-OK, though I'm not sure how it relates to handling direct writes to core WP tables (aren't all the refund-related writes here to COT tables)?

Reply:
Actually, this PR was causing a bug with the creation of a refund due to a type of declaration conflict. I added this test since the bug fix was also pushed to this branch.